app/core/src/test/java/com/fsck/k9/cache/InMemoryCacheTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.fsck.k9.cache
+
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertNull
+import kotlin.test.assertSame
+import kotlin.test.assertTrue
+
+class InMemoryCacheTest {
```

Comment:
There doesn't seem to be a meaningful implementation detail `InMemoryCacheTest` could test. To test the contract, I suggest changing it to a parameterized test that will be used for all implementations. Then `ExpiringCacheTest` could be limited to test the expiry behavior.

```kotlin
data class CacheTestData(val name: String, val createCache: () -> Cache<String, String>) {
    override fun toString() = name
}

@RunWith(Parameterized::class)
class CacheTest(data: CacheTestData) {
    private val cache = data.createCache()

    companion object {
        @JvmStatic
        @Parameterized.Parameters(name = "{0}")
        fun data(): Collection<CacheTestData> {
            return listOf(
                CacheTestData("InMemoryCache") { InMemoryCache() },
                CacheTestData("ExpiringCache") { ExpiringCache(InMemoryCache()) },
                CacheTestData("SynchronizedCache") { SynchronizedCache(InMemoryCache()) }
            )
        }
    }

    @Test
    fun `write to cache then read`() {
        cache[KEY] = VALUE

        assertThat(cache[KEY]).isEqualTo(VALUE)
    }

   // …
}
```